kernel/src/main/java/org/kframework/utils/options/OuterParsingOptions.java

```diff
@@ -51,4 +51,7 @@ public synchronized File mainDefinitionFile(FileUtil files) {
 
     @Parameter(names="--md-selector", description="Preprocessor: for .md files, select only the md code blocks that match the selector expression. Ex:'k&(!a|b)'.")
     public String mdSelector = "k";
+
+    @Parameter(names="--profile-rule-parsing", description="Generate time in seconds to parse each rule in the semantics. Found in -kompiled directory under timing.log.")
+    public String profileRules;
```

Comment:
This is not an appropriate place for this flag because it does not relate to outer parsing.

Reply:
The only alternative would be to make an entirely new `InnerParsingOptions.java` class and add it to kompile and kprove (not sure if it's needed anywhere else). Is this what you have in mind?

Reply:
that would probably be better. Right now, this flag is being included in KDep, which is completely inappropriate.